modules/solid_properties/src/userobjects/ThermalUCProperties.C

```diff
@@ -0,0 +1,83 @@
+#include "ThermalUCProperties.h"
+#include "libmesh/utility.h"
+
+registerMooseObject("SolidPropertiesApp", ThermalUCProperties);
+
+InputParameters
+ThermalUCProperties::validParams()
+
+{
+  InputParameters params = ThermalSolidProperties::validParams();
+
+  params.addRangeCheckedParam<Real>("density", 13824.7, "density > 0.0", "(Constant) density");
+  params.addClassDescription("Uranium Carbide (UC) thermal properties.");
+  return params;
+}
+
+ThermalUCProperties::ThermalUCProperties(const InputParameters & parameters)
+  : ThermalSolidProperties(parameters), _rho_const(getParam<Real>("density"))
+{
+}
+
+Real
+ThermalUCProperties::cp_from_T(const Real & T) const
+{
+  return 239.7 - 5.068e-3 * T + 1.7604e-5 * Utility::pow<2>(T) - 3488100 / Utility::pow<2>(T);
```

Comment:
add an if on range of validity
```
298 K $\le$ T $\le$ 2838 K
```
and declare solution is invalid outside. 

You may need to add the SOlutionInvalidInterface first to the classes that the SolidFluidProperties first
Please let me know if you need help with that

Reply:
https://github.com/idaholab/moose/blob/next/modules/fluid_properties/src/fluidproperties/NaKFluidProperties.C
for an example of doing this